src/System.IO.FileSystem/src/System/IO/FileSystemInfo.Unix.cs

```diff
@@ -228,7 +228,9 @@ void IFileSystemObject.Refresh()
         {
             // This should not throw, instead we store the result so that we can throw it
             // when someone actually accesses a property.
-            int result = Interop.Sys.Stat(FullPath, out _fileStatus);
+            // Use LStat (rather than Stat) so that information about a symbolic link will be retrieved
+            // (rather than information about the target) if FullPath points to a symbolic link.
+            int result = Interop.Sys.LStat(FullPath, out _fileStatus);
```

Comment:
This will change other properties (such as file size and attributes like 'hidden') to report information on the symlink instead of the target file, also.

That seems desirable, though, because it looks like that matches the Windows behavior for those properties (FileInfo.Size on a sym link returns 0, for example).